exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -15,134 +15,108 @@
+   * TraceConfiguration} object that requires no input from the user. Since no project ID is
+   * specified, default project ID is used instead. See {@link ServiceOptions#getDefaultProjectId()}
+   * for details.
+   *
+   * <p>This method defers the creation of an actual {@link TraceExporter} to a point when it is
+   * actually needed - which is when the spans need to be exported. As a result, while this method
+   * does not throw any exception, an exception may still be thrown during the attempt to generate
+   * the actual {@link TraceExporter}.
+   *
+   * @return A configured instance of {@link TraceExporter} which gets initialized lazily once
+   *     {@link TraceExporter#export(Collection)} is called.
+   */
+  public static SpanExporter createWithDefaultConfiguration() {
+    return generateSkeletonTraceExporter(DEFAULT_CONFIGURATION);
   }
 
-  TraceExporter(
-      String projectId,
-      CloudTraceClient cloudTraceClient,
-      ImmutableMap<String, String> attributeMappings,
-      Map<String, AttributeValue> fixedAttributes) {
-    this.projectId = projectId;
-    this.cloudTraceClient = cloudTraceClient;
-    this.projectName = ProjectName.of(projectId);
-    this.translator = new TraceTranslator(attributeMappings, fixedAttributes);
+  /**
+   * Method that generates an instance of {@link TraceExporter} using a {@link TraceConfiguration}
+   * that allows the user to provide custom configuration for Traces.
+   *
+   * <p>This method defers the creation of an actual {@link TraceExporter} to a point when it is
+   * actually needed - which is when the spans need to be exported. As a result, while this method
+   * does not throw any exception, an exception may still be thrown during the attempt to generate
+   * the actual {@link TraceExporter}.
+   *
+   * @param configuration The {@link TraceConfiguration} object that determines the user preferences
+   *     for Trace.
+   * @return An instance of {@link TraceExporter} as a {@link SpanExporter} object
+   */
+  public static SpanExporter createWithConfiguration(TraceConfiguration configuration) {
+    return generateSkeletonTraceExporter(configuration);
   }
 
   @Override
   public CompletableResultCode flush() {
-    // We do no exporter buffering of spans, so we're always flushed.
-    return CompletableResultCode.ofSuccess();
+    if (internalTraceExporter.get() == null) {
+      return CompletableResultCode.ofFailure();
+    }
+    return internalTraceExporter.get().flush();
   }
 
   @Override
-  public CompletableResultCode export(Collection<SpanData> spanDataList) {
-    List<Span> spans = new ArrayList<>(spanDataList.size());
-    for (SpanData spanData : spanDataList) {
-      spans.add(translator.generateSpan(spanData, projectId));
+  public CompletableResultCode export(@Nonnull Collection<SpanData> spanDataList) {
+    if (Objects.isNull(internalTraceExporter.get())) {
+      synchronized (this) {
```

Comment:
I'm confused by the synchronization model here - usually an `AtomicReference` is used to avoid synchronization. You might be able to avoid a lot of this complexity by changing `internalTraceExporter` to `Suppliers.memoized(() -> new InternalTraceExporter(this.config))` and making it do all the work.